pkg/reconciler/kubernetes/tektonchains/tektonchains.go

```diff
@@ -0,0 +1,406 @@
+			// to allow changes to get deployed
+			r.enqueueAfter(tc, 20*time.Second)
+			return nil
+		}
+	}
+
+	// Mark InstallerSetAvailable
+	tc.Status.MarkInstallerSetAvailable()
+
+	ready := installedTIS.Status.GetCondition(apis.ConditionReady)
+	if ready == nil {
+		tc.Status.MarkInstallerSetNotReady("Waiting for installation")
+		r.enqueueAfter(tc, 10*time.Second)
+		return nil
+	}
+
+	if ready.Status == corev1.ConditionUnknown {
+		tc.Status.MarkInstallerSetNotReady("Waiting for installation")
+		r.enqueueAfter(tc, 10*time.Second)
+		return nil
+	} else if ready.Status == corev1.ConditionFalse {
+		tc.Status.MarkInstallerSetNotReady(ready.Message)
+		r.enqueueAfter(tc, 10*time.Second)
+		return nil
+	}
+
+	// Mark InstallerSet Ready
+	tc.Status.MarkInstallerSetReady()
+
+	if err := r.extension.PostReconcile(ctx, tc); err != nil {
+		tc.Status.MarkPostReconcilerFailed(fmt.Sprintf("PostReconciliation failed: %s", err.Error()))
+		return err
+	}
+
+	// Mark PostReconcile Complete
+	tc.Status.MarkPostReconcilerComplete()
+
+	// Update the object for any spec changes
+	if _, err := r.operatorClientSet.OperatorV1alpha1().TektonChainses().Update(ctx, tc, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FinalizeKind removes all resources after deletion of a TektonChains.
+func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.TektonChains) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	// Delete CRDs before deleting rest of resources so that any instance
+	// of CRDs which has finalizer set will get deleted before we remove
+	// the controller's deployment for it
+	if err := r.manifest.Filter(mf.CRDs).Delete(); err != nil {
+		logger.Error("Failed to deleted CRDs for TektonChains")
+		return err
+	}
+
+	labelSelector, err := common.LabelSelector(ls)
+	if err != nil {
+		return err
+	}
+	if err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+		DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: labelSelector,
+		}); err != nil {
+		logger.Error("Failed to delete installer set created by TektonChains", err)
+		return err
+	}
+
+	if err := r.extension.Finalize(ctx, original); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) updateTektonChainsStatus(ctx context.Context, tc *v1alpha1.TektonChains, createdIs *v1alpha1.TektonInstallerSet) error {
+	// update the tc with TektonInstallerSet and releaseVersion
+	tc.Status.SetTektonInstallerSet(createdIs.Name)
+	tc.Status.SetVersion(r.chainsVersion)
+
+	// Update the status with TektonInstallerSet so that any new thread
+	// reconciling knows that TektonInstallerSet is created otherwise
+	// there will be 2 instances created if we don't update status here
+	if _, err := r.operatorClientSet.OperatorV1alpha1().TektonChainses().
+		UpdateStatus(ctx, tc, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
```

Comment:
we can skip this update call now right? 🤔
as we are not dependent on installer name in status 
so we can just set it and return ... and let knative handle the update status part which it does

Reply:
this can be handled in further pr too ..  :)